content/en/docs/migrating/_index.md

```diff
@@ -0,0 +1,819 @@
+[endpoints.Resolver]({{< apiref v1="aws/endpoints#Resolver" >}}) this must be migrated to conform to the
+`aws.EndpointResolver` interface. [aws.EndpointResolverFunc]({{< apiref "aws#EndpointResolverFunc" >}}) is provided as
+a convenient way to wrap a resolver function to satisfy the `aws.EndpointResolver` interface.
+
+For more information on endpoints and implementing a custom resolver, see
+[Configuring Client Endpoints]({{% ref "/docs/configuring-sdk/endpoints.md" %}}).
+
+### Invoking API Operations
+
+The number of service client operations methods have been reduced down significantly. The `<OperationName>Request`,
+`<OperationName>dWithContext`, and `<OperationName>` methods have all been consolidated into a single operation method.
+
+#### Example
+
+This example shows how calls to the {{% alias service=s3 %}} PutObject operation would be migrated from AWS SDK for Go
+to {{% alias sdk-go %}}.
+
+##### V1
+```go
+import "context"
+import "github.com/aws/aws-sdk-go/service/s3"
+
+// ...
+
+client := s3.New(sess)
+
+// Pattern 1
+output, err := client.PutObject(&s3.PutObjectInput{
+	// input parameters
+})
+
+// Pattern 2
+output, err := client.PutObjectWithContext(context.TODO(), &s3.PutObjectInput{
+    // input parameters
+})
+
+// Pattern 3
+req, output := client.PutObjectRequest(context.TODO(), &s3.PutObjectInput{
+    // input parameters
+})
+err := req.Send()
+```
+
+##### V2
+
+```go
+import "context"
+import "github.com/aws/aws-sdk-go-v2/service/s3"
+
+// ...
+
+client : s3.NewFromConfig(cfg)
+
+output, err := client.PutObject(context.TODO(), &s3.PutObjectInput{
+	// input parameters
+})
+```
+
+### Service Data Types
+
+The top-level input and output types of an operation are found in the service client package. The input and output shape
+for a given operation will always follow the pattern of `<OperationName>Input` and `<OperationName>Output`. Where
+`OperationName` is the name of the operation you are invoking. For example the input and output shape for
+the {{% alias service=S3 %}} PutObject operation are [PutObjectInput]({{< apiref "service/s3#PutObjectInput" >}}) and
+[PutObjectOutput]({{< apiref "service/s3#PutObjectOutput" >}}) respectively.
+
+All other service data types other then the input and output types have been migrated to the `types` package located
+under the service client package import path hierarchy. For example, the
+[s3.AccessControlPolicy]({{< apiref v1="service/s3#AccessControlPolicy" >}}) type is now located at
+[types.AccessControlPolicy]({{< apiref "service/s3/types#AccessControlPolicy" >}}).
+
+#### Enumeration Values
+
+The SDK now provides a typed experience for all API enumeration fields. Rather than using a string literal value copied
+from the service API reference documentation, you can now use one of the concrete types found in the service client's
+`types` package. For example {{% alias service=S3 %}} PutObjectInput operation can be provided an optional canned ACL
```

Comment:
For example the -> For example, you can provide the
can be provided an optional canned -> with an